src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -963,7 +963,7 @@ public function __construct()
 
 EOF;
         if (null !== $this->targetDirRegex) {
```

Comment:
in which case can this regex be null ? The `$this->containerDir` variable will not be assigned at all currently in this case.

Reply:
OK, looking at it, it might happens in some cases. So I suggest you to move the `$this->containerDir = $dir` assignment to a dedicated statement added in the compiled code outside this `if`

Reply:
indeed, wrong "if", fixed thanks